osf/models/user.py

```diff
@@ -1802,31 +1802,27 @@ def get_activity_points(self):
         return analytics.get_total_activity_count(self._id)
 
     def get_or_create_cookie(self, secret=None):
-        """Find the cookie for the given user
-        Create a new session if no cookie is found
+        """Find the cookie from the most recent session for the given user. Create a new session, compute its
+        cookie value using the default or provide secret, and return the new cookie if no existing session is found.
 
         :param str secret: The key to sign the cookie with
         :returns: The signed cookie
         """
         secret = secret or website_settings.SECRET_KEY
-
         user_session_map = UserSessionMap.objects.filter(user__id=self.id).order_by('-expire_date').first()
-
-        if not SessionStore().exists(session_key=user_session_map.session_key):
+        if not user_session_map or not SessionStore().exists(session_key=user_session_map.session_key):
             user_session = SessionStore()
-            user_session.create()
             user_session['auth_user_id'] = self._id
             user_session['auth_user_username'] = self.username
             user_session['auth_user_fullname'] = self.fullname
-            user_session.save()
+            user_session.create()
             UserSessionMap.objects.create(
                 user=self,
                 session_key=user_session.session_key,
                 expire_date=user_session.get_expiry_date()
             )
         else:
             user_session = SessionStore(session_key=user_session_map.session_key)
```

Comment:
Maybe refactor this to something like:
```
user_session = SessionStore()
if user_session_map and SessionStore().exists(session_key=user_session_map.session_key):
        user_session = SessionStore(session_key=user_session_map.session_key)
    else:
        user_session.create()
        user_session['auth_user_id'] = self._id
        user_session['auth_user_username'] = self.username
        user_session['auth_user_fullname'] = self.fullname
        user_session.save()
        UserSessionMap.objects.create(
            user=self,
            session_key=user_session.session_key,
            expire_date=user_session.get_expiry_date()
        )
```

Reply:
Two notes:
* Why defining `user_session = SessionStore()` if this is only used in the `else`
* And should we avoid `.create()`->setting data->`.save()` by calling `.create()` after setting data?

Reply:
I still need to revert `if ... else ...` so that the condition is shorter and that `if ... else ...` looks better.

Reply:
It's okay if we don't do `user_session = SessionStore()` and call `create()` after setting session data. But `not statement1 or not statement2` is a bit counter-intuitive for me.